matchms/similarity/BaseSimilarityFunction.py

```diff
@@ -0,0 +1,54 @@
+from typing import List
+import numpy
+from matchms.typing import SpectrumType
+
+
+class BaseSimilarityFunction:
+    """Similarity function base class.
+    When building a custom similarity measure, inherit from this class and implement
+    the desired methods.
+    """
+    # Set key characteristics as class attributes
+    is_commutative = True
```

Comment:
Property is now not shown in Sphinx, add attributes section to document
```suggestion
    Attributes
    ----------
    is_commutative
       Whether similarity function is commutative, aka order of spectrums does not matter. Default is True.
    """
    is_commutative = True
```

Reply:
Good point, thanks! Done.